sdk/src/main/java/co/decodable/sdk/pipeline/DecodableSecret.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright Decodable, Inc.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package co.decodable.sdk.pipeline;
+
+import co.decodable.sdk.pipeline.exception.SecretNotFoundException;
+import co.decodable.sdk.pipeline.internal.DecodableSecretImpl;
+import co.decodable.sdk.pipeline.util.Incubating;
+import java.time.Instant;
+
+// spotless:off
+/**
+ * Represents a <a href="https://docs.decodable.co/docs/manage-secrets">Decodable secret</a>.
+ * Exposes both metadata and the secret value. Create an instance and access its properties like
+ * this:
+ *
+ * <p>
+ * {@snippet :
+   var mySecret = DecodableSecret.withName("my-secret");
+   var mySecretValue = mySecret.value(); doStuffWithMySecret(mySecretValue);
+   }
```

Comment:
Ah, I was hoping for something more realistic, with an actual connector example. Challenge is it would have to be one which we don't support first-class managed in Decodable while still being relatable for many readers. Maybe it's not a good idea, dunno.

Reply:
I was afraid you would say that 🙈.

Not sure I really like the idea, though. The secret class doesn't really have anything to do with specific connectors, and we'd also run the risk of having an outdated JavaDoc if something in the connector changes (e.g. required properties). Moreover, we don't have anything about connectors in the SDK, do we? So we'd have to write lower-level Java/Flink code to connect to some imaginary connection, which feels overkill to me (and too much for a JavaDoc here). But I'll check with the connector folks for ideas.
 
If we want to have an actual example, I'd prefer to have a complete one similar to `DataStreamJob`. It would have to be one where it's easy to set up the connector in Decodable though, so people could actually set up the resources and run the example quickly.

Reply:
> It would have to be one where it's easy to set up the connector in Decodable though

Not quite following on that one. One use case for the SDK--and AFAIU, the motivator for secrets support--is to enable usage of connectors which are _not_ supported as managed connectors on Decodable (yet). So I think this is what we somehow should show in docs (be it JavaDoc or elsewhere).

Reply:
So then it would only be theoretical code and users would not be able to actually run it anywhere, right (unless they have an external connection of this type and adapt the parameters)?

Do you happen to know an example of a not yet supported connector that would be interesting to use, and some Flink code to go with it? 😇 

Another use case is to combine the connector with some processing or other logic, which would also make sense for supported connectors. But this may be less interesting than not yet supported connectors.

Reply:
> So then it would only be theoretical code and users would not be able to actually run it anywhere, right (unless they have an external connection of this type and adapt the parameters)?

Yes, exactly.

> Do you happen to know an example of a not yet supported connector that would be interesting to use, and some Flink code to go with it? 😇

The [SQL Server CDC connector](https://ververica.github.io/flink-cdc-connectors/master/content/connectors/sqlserver-cdc.html) would be a good example, e.g. like so:

```
SourceFunction<String> sourceFunction = SqlServerSource.<String>builder()
      .hostname("localhost")
      .port(1433)
      .database("inventory")
      .tableList("dbo.items")
      .username("my-sql-server-user")
      .password(DecodableSecret.withName("my-sql-server-password").value())
      .deserializer(new JsonDebeziumDeserializationSchema())
      .build();
```

Reply:
Thanks. Put this in the JavaDoc instead of the dummy example.